fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -0,0 +1,221 @@
+    private static long oldInitialDelayMillis;
+    private static int oldMaxAttempts;
+
+    @BeforeAll
+    public static void setUpForClass() {
+        oldInitialDelayMillis = FDBDatabaseFactory.instance().getInitialDelayMillis();
+        FDBDatabaseFactory.instance().setInitialDelayMillis(2L);
+        oldMaxDelayMillis = FDBDatabaseFactory.instance().getMaxDelayMillis();
+        FDBDatabaseFactory.instance().setMaxDelayMillis(4L);
+        oldMaxAttempts = FDBDatabaseFactory.instance().getMaxAttempts();
+        FDBDatabaseFactory.instance().setMaxAttempts(100);
+    }
+
+    @AfterAll
+    public static void tearDownForClass() {
+        FDBDatabaseFactory.instance().setMaxDelayMillis(oldMaxDelayMillis);
+        FDBDatabaseFactory.instance().setInitialDelayMillis(oldInitialDelayMillis);
+        FDBDatabaseFactory.instance().setMaxAttempts(oldMaxAttempts);
+    }
+
+    @BeforeEach
+    public void setUp() {
+        fdb.run(context -> {
+            FDBRecordStore.deleteStore(context, path);
+            return null;
+        });
+    }
+
+    private void rebuildIndexMetaData(final FDBRecordContext context, final String document, final Index index) {
+        Pair<FDBRecordStore, QueryPlanner> pair = LuceneIndexTestUtils.rebuildIndexMetaData(context, path, document, index, useRewritePlanner);
+        this.recordStore = pair.getLeft();
+        this.planner = pair.getRight();
+    }
+
+    private void disableIndex(Index index, String document) {
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, document, index);
+            recordStore.markIndexDisabled(index).join();
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void luceneOnlineIndexingTestSimple(boolean preventMergeDuringIndexing) {
+        Index index = SIMPLE_TEXT_SUFFIXES;
+        disableIndex(index, SIMPLE_DOC);
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            recordStore.saveRecord(createSimpleDocument(2222L, WAYLON + " who?", 1));
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            recordStore.saveRecord(createSimpleDocument(2222L, WAYLON + " who?", 1));
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
```

Comment:
For the number of records in the store, wouldn't it be expected that this will index the entire store in a single transaction? It seems like it would be better to force it to have multiple transactions during indexing.

Reply:
✅ 
Added `transactionLimit` param to the `luceneOnlineIndexingTestAny` test.